### PR TITLE
Fd 1059 circle ci

### PR DESCRIPTION
**Pull request by Emyrk**
Added circle ci. Needed to make some dependency adjustments to pass unit tests.

There are some function definition changes in this that needed to be updated in factomd to enable all unit tests to compile + pass.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,78 @@
+# Golang CircleCI 2.0 configuration file
+#
+# Check https://circleci.com/docs/2.0/language-go/ for more details
+version: 2
+jobs:
+  build:
+    docker:
+      # specify the version
+      - image: circleci/golang:1.10
```
Comment (carryforward) [inline .circleci/config.yml]: lets do 1.12

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+
+      - persist_to_workspace:
+          root: /tmp
+          paths: go
+
+  test:
+    working_directory: /tmp # All the binaries are saved here
+    docker:
+      - image: circleci/golang:1.10
```
Comment (carryforward) [inline .circleci/config.yml]: 1.12 as well

File: glide.lock [hunk 3]
```diff
@@ -211,7 +215,7 @@ imports:
   - unicode/bidi
   - unicode/norm
 - name: google.golang.org/genproto
-  version: a7e196e89fd3a3c4d103ca540bd5dac3a736e375
+  version: 32ee49c4dd805befd833990acba36cb75042378c
```
Comment (carryforward) [inline glide.lock]: this changes from the latest (a7e19) to one from a year ago (32ee49)

Comment (Emyrk) [reply glide.lock]: hmm, that was glide update choosing that. I'll get it to the latest verison

Comment (carryforward) [reply glide.lock]: sigh, when steven updated this to the latest one, his tests would not pass.

Comment (Emyrk) [reply glide.lock]: This dep is not in any core functionality, only used during the test suite.

Updating it breaks said tests

**Pull request merged**